bin/nvcc_wrapper

```diff
@@ -242,6 +242,77 @@ do
     cuda_args="$cuda_args $1 $2"
     shift
     ;;
+  # Handle Werror. Note, we must differentiate between the ones going to nvcc and the host compiler
+  # --Werror kind,... OR --Werror=kind,... <- always to nvcc
+  --Werror)
+    cuda_args="$cuda_args $1 $2"
+    shift
+    ;;
+  --Werror=*)
+    cuda_args="$cuda_args $1"
+    ;;
+  # -Werror kind,... where kind is one of {all-warnings, cross-execution-space-call, reorder, default-stream-launch, missing-launch-bounds, ext-lambda-captures-this, deprecated-declarations} <- goes to nvcc
+  # -Werror not followed by any kind as mentioned above goes to host compiler without any arguments
+  -Werror)
+    if [ $# -gt 1 ]; then
+      kinds=(${2//,/ })
```

Comment:
Shellcheck warnings for the added code:
```
[Line 258:](javascript:setPosition(258, 14))
      kinds=(${2//,/ })
             ^-- [SC2206](https://github.com/koalaman/shellcheck/wiki/SC2206) (warning): Quote to prevent word splitting/globbing, or split robustly with mapfile or read -a.
 
[Line 279:](javascript:setPosition(279, 16))
    kinds_str=(${1:8}) # strip -Werror=
               ^-- [SC2206](https://github.com/koalaman/shellcheck/wiki/SC2206) (warning): Quote to prevent word splitting/globbing, or split robustly with mapfile or read -a.
 
[Line 280:](javascript:setPosition(280, 12))
    kinds=(${kinds_str//,/ })
           ^-- [SC2206](https://github.com/koalaman/shellcheck/wiki/SC2206) (warning): Quote to prevent word splitting/globbing, or split robustly with mapfile or read -a.
```